lfs_ffi.lua

```diff
@@ -186,24 +186,23 @@ local function iterator(dir)
     end
 end
 
-local dir_obj_type = ffi.metatype([[
-    struct {
-        DIR *_dentry;
-        bool closed;
-    }
-]],
-{__index = {
+
+local dir_obj_mt = {__index = {
     next = iterator,
     close = close,
 }, __gc = close
```

Comment:
I think we need to make the `__gc` work as user can bypass the `close` with calling `next` directly.

Reply:
Could you add a test like this one: https://github.com/spacewander/luafilesystem/blob/10831d3de10acaa28ff58e939ed0f7c4f4d2424f/lfs_spec.lua#L200 but only iterate for one or two times?